src/mugs.js

```diff
@@ -78,8 +78,119 @@ define([
         getIcon: function () {
             return this.options.getIcon(this);
         },
+        /**
+         * Validate mug
+         *
+         * This method may fire a "messages-changed" event.
+         *
+         * @param attr - The property to validate. All properties will
+         *      be validated if this argument is omitted.
+         * @returns - True if validation messages changed else false.
+         */
+        validate: function (attr) {
+            var mug = this,
+                changed;
+            this.form.updateLogicReferences(this, attr);
+            if (attr) {
+                changed = this._validate(attr);
```

Comment:
Small nitpick, but it seems slightly more confusing to have `this._validate` here and `mug._validate` later in same function when they are the same thing